### PR TITLE
Nodeport loadbalanceing

### PR DESCRIPTION
**Pull request by alillistone**
Enables customers to explose NodePort services using OpenStack Octavia loadbalancer

### CODE REVIEW

**Comment by gellner** [comment]
So, @alillistone can't approve his own PR but I've added a load of crap to it:

- Add ClusterID-based query to pick out the LB (is this needed? Maybe not)
- Add handling for empty CM or malformed CM
- Remove pip-requirements.txt - have tested using the one in openshift-upi-ansible and we should keep them the same.
- Create the ukc-ingress namespace earlier

**Review by gellner — APPROVED** [review]
LGTM now, but I'm open to suggestions/improvements from @benbacon

File: post-deployment/openstack/nodeport-loadbalancing-ingress/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+**WARNING**: This is not supported for OpenStack Projects with multiple clusters.
+
+## Prerequisite
+
+Before you run this, you must first have a loadbalancer deployed in your OpenStack Environment. This can be done by running the following playbooks in order:
+
+nodeport-loadbalance-network.yaml
+nodeport-loadbalance-ingress.yaml
+
+The can be found in https://github.com/UKCloud/openshift-upi-ansible/tree/nodeport-loadbalancing
```
Comment (benbacon) [inline post-deployment/openstack/nodeport-loadbalancing-ingress/README.md:14]: Should we be referencing a branch that is likely going to be deleted once merged?

File: post-deployment/openstack/nodeport-loadbalancing-ingress/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# NodePort loadbalancing using OpenStack Octavia
+
+This features enables customers to expose their NodePorts via a single public IP using OpenStack Octavia Loadbalancer.
```
Comment (benbacon) [inline post-deployment/openstack/nodeport-loadbalancing-ingress/README.md:3]:
```suggestion
This feature enables customers to expose their NodePorts via a single public IP using OpenStack Octavia Loadbalancer.
```

File: post-deployment/openstack/nodeport-loadbalancing-ingress/build/playbooks/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export K8S_AUTH_VERIFY_SSL=no
+
+# Add uid to /etc/passwd to allow Ansible to run
+if [ `id -u` -ge 500 ]; then
+    echo "runner:x:`id -u`:`id -g`:,,,:/runner:/bin/bash" > /tmp/passwd
+    cat /tmp/passwd >> /etc/passwd
+    rm /tmp/passwd
+fi
+
+ansible-playbook /usr/local/playbooks/update-nodeport-loadbalancer-members.yaml
```
Comment (benbacon) [inline post-deployment/openstack/nodeport-loadbalancing-ingress/build/playbooks/entrypoint.sh:10]:
```suggestion
ansible-playbook /usr/local/playbooks/update-nodeport-loadbalancer-members.yaml

```

File: post-deployment/openstack/nodeport-loadbalancing-ingress/README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    - nodeport: 30418
+      listener: 1234
+      allowed_ips:
+        - 0.0.0.0/0
+        - 80.81.82.83/24
+```
+
+nodeport = the port that the NodePort service is exposing on. 
+listener = the port that you want the loadbalancer to listen on.
+allowed_ips = list of source CIDR's that you want the loadbalanacer to only accept traffic from.
```
Comment (benbacon) [inline post-deployment/openstack/nodeport-loadbalancing-ingress/README.md:41]:
```suggestion
allowed_ips = list of source CIDRs that you want the loadbalancer to only accept traffic from.
```

File: post-deployment/openstack/nodeport-loadbalancing-ingress/README.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+data:
+  openstack: |
+    - nodeport: 30418
+      listener: 1234
+      allowed_ips:
+        - 0.0.0.0/0
+        - 80.81.82.83/24
+```
+
+nodeport = the port that the NodePort service is exposing on.
```
Comment (benbacon) [inline post-deployment/openstack/nodeport-loadbalancing-ingress/README.md:39]:
```suggestion
nodeport = the port that the NodePort service is exposed on.
```

**Pull request merged**